chains/evm/contracts/libraries/MessageV1Codec.sol

```diff
@@ -332,115 +381,114 @@ library MessageV1Codec {
   /// @return message The decoded MessageV1 struct.
   function _decodeMessageV1(
     bytes calldata encoded
-  ) internal pure returns (MessageV1 memory) {
-    if (encoded.length < 37) revert InvalidDataLength(EncodingErrorLocation.MESSAGE_MIN_SIZE); // Minimum size check.
-
-    MessageV1 memory message;
-    uint256 offset = 0;
-
-    uint8 version = uint8(encoded[offset++]);
-    if (version != 1) revert InvalidEncodingVersion(version);
-
-    // Protocol Header.
-    // sourceChainSelector (8 bytes, big endian).
-    message.sourceChainSelector = uint64(bytes8(encoded[offset:offset + 8]));
-    offset += 8;
```

Comment:
You can see the pointer updates that we can skip when we have the static length fields first